src/deadline/job_attachments/api/manifest.py

```diff
@@ -0,0 +1,143 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+import datetime
+import os
+from typing import List, Optional, Tuple
+
+
+from deadline.client.api._job_attachment import _hash_attachments
+from deadline.client.cli._common import _ProgressBarCallbackManager
+from deadline.client.cli._groups.click_logger import ClickLogger
+from deadline.job_attachments._diff import compare_manifest
+from deadline.job_attachments._glob import _process_glob_inputs, _glob_paths
+from deadline.job_attachments.asset_manifests.base_manifest import (
+    BaseManifestPath,
+)
+from deadline.job_attachments.asset_manifests.decode import decode_manifest
+from deadline.job_attachments.exceptions import ManifestCreationException
+from deadline.job_attachments.models import (
+    FileStatus,
+    GlobConfig,
+    JobAttachmentS3Settings,
+    ManifestSnapshot,
+    default_glob_all,
+)
+from deadline.job_attachments.upload import S3AssetManager
+
+"""
+APIs here should be business logic only. It should perform one thing, and one thing well. 
+It should use basic primitives like S3 upload, download, boto3 APIs.
+These APIs should be boto3 session agnostic and a specific Boto3 Credential to use.
+"""
+
+
+def _manifest_snapshot(
+    root: str,
+    destination: str,
+    name: str,
+    include: Optional[List[str]] = None,
+    exclude: Optional[List[str]] = None,
+    include_exclude_config: Optional[str] = None,
+    diff: Optional[str] = None,
+    logger: ClickLogger = ClickLogger(False),
```

Comment:
nit - should this be 
```suggestion
    logger: Optional[ClickLogger] = ClickLogger(False),
```

Reply:
No - we should always have a logger! The false means it does standard out print.